core/src/main/java/com/linecorp/armeria/client/endpoint/CompositeEndpointGroup.java

```diff
@@ -67,11 +68,12 @@
             });
         }
 
-        initialEndpointsFuture =
-                CompletableFuture.anyOf(this.endpointGroups.stream()
-                                                           .map(EndpointGroup::whenReady)
-                                                           .toArray(CompletableFuture[]::new))
-                                 .thenApply(unused -> endpoints());
+        CompletableFuture.anyOf(this.endpointGroups.stream()
+                                                   .map(EndpointGroup::whenReady)
+                                                   .toArray(CompletableFuture[]::new))
+                         .thenAccept(unused -> {
+                             initialEndpointsFuture.complete(new LazyList<>(this::endpoints));
```

Comment:
What happens if the future returned by `anyOf()` completes exceptionally?

Reply:
Thanks fixed. 😉